pkg/asset/manifests/openstack/cluster.go

```diff
@@ -35,33 +35,42 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		Spec: capo.OpenStackClusterSpec{
 			CloudName: cloudName,
 			IdentityRef: &capo.OpenStackIdentityReference{
-				Kind: "Secret",
 				Name: clusterID.InfraID + "-cloud-config",
 			},
 			// We disable management of most networking resources since either
 			// we (the installer) will create them, or the user will have
 			// pre-created them as part of a "Bring Your Own Network (BYON)"
 			// configuration
-			ManagedSecurityGroups:      false,
+			ManagedSecurityGroups:      nil,
 			DisableAPIServerFloatingIP: true,
 			// TODO(stephenfin): update when we support dual-stack (there are
 			// potentially *two* IPs here)
-			APIServerFixedIP:  openstackInstallConfig.APIVIPs[0],
-			DNSNameservers:    openstackInstallConfig.ExternalDNS,
-			ExternalNetworkID: openstackInstallConfig.ExternalNetwork,
+			APIServerFixedIP: openstackInstallConfig.APIVIPs[0],
+			ExternalNetwork:  capo.NetworkFilter{ID: openstackInstallConfig.ExternalNetwork},
 			Tags: []string{
 				fmt.Sprintf("openshiftClusterID=%s", clusterID.InfraID),
 			},
 		},
 	}
 	if openstackInstallConfig.ControlPlanePort != nil {
 		// TODO(maysa): update when BYO dual-stack is supported in CAPO
-		openStackCluster.Spec.Network.ID = openstackInstallConfig.ControlPlanePort.Network.ID
-		openStackCluster.Spec.Network.Name = openstackInstallConfig.ControlPlanePort.Network.Name
-		openStackCluster.Spec.Subnet.ID = openstackInstallConfig.ControlPlanePort.FixedIPs[0].Subnet.ID
-		openStackCluster.Spec.Subnet.Name = openstackInstallConfig.ControlPlanePort.FixedIPs[0].Subnet.Name
+		openStackCluster.Spec.Network = capo.NetworkFilter{
+			Name: openstackInstallConfig.ControlPlanePort.Network.Name,
+			ID:   openstackInstallConfig.ControlPlanePort.Network.ID,
+		}
+		openStackCluster.Spec.Subnets = []capo.SubnetFilter{
```

Comment:
If there there is more than one `openstackInstallConfig.ControlPlanePort.FixedIPs` we should add all of them.

Also, can you remove the TODO?